### PR TITLE
Add support for custom properties (e.g. external billing, subject length)

### PR DESCRIPTION
**Pull request by cmaier**
Custom properties can now be added and implemented easier. As an example the subject text edit for the external billing has been make configurable using a `CustomProperty`.

APPS-1491 _(APPS-1497)_

### How to test?

* Checkout this branch for an app making use of external billing, limit the subject length and check if it works.

### Definition of Done

- [x] Changelog gepflegt
- [x] Dokumentation gepflegt
- [x] Alle Anforderung des Issues sind erfüllt
- [x] Selbst greviewed _(aka [Self-Reviews (eng)](https://www.nerdwallet.com/blog/engineering/why-you-should-be-doing-self-reviews/))_
- [ ] Review mit Product Owner _(Release-Variante o. Minified Build)_

#### App Tests
- ~~Minified getestet? _(aka. Release Build)_~~
- ~~Environements beachtet _(Production/Staging)_~~
- ~~Unterstützte Sprachen sind getestet~~
- ~~Light-/Dark-Mode getestet~~
- [x] Edge-Cases getestet
- [x] Android API Levels wurden berücksichtigt _(minSdk?)_

#### Testing
- ~~Tests geschrieben _(aka Unit-Tests, Integration-Tests)_~~
- [x] Tests lokal laufen gelassen

### CODE REVIEW

File: ui/src/main/java/io/snabble/sdk/ui/payment/externalbilling/ui/widgets/SubjectAlertDialog.kt [hunk 1]
```diff
@@ -15,7 +16,7 @@ import com.google.android.material.button.MaterialButton
 import com.google.android.material.textfield.TextInputLayout
 import io.snabble.sdk.ui.R
 
-class SubjectAlertDialog(context: Context) : Dialog(context) {
+class SubjectAlertDialog(context: Context, private val maxSubjectLength: Int = -1) : Dialog(context) {
```
Comment (Fabtron) [inline ui/src/main/java/io/snabble/sdk/ui/payment/externalbilling/ui/widgets/SubjectAlertDialog.kt:19]: Is there a reason using `-1` as default? if not I would suggest using a nullable value make the overall usage cleaner like:

```kotlin
class SubjectAlertDialog(context: Context, private val maxSubjectLength: Int? = null) : Dialog(context) { ...
```
```kotlin
editTextField.filters = arrayOf(InputFilter.LengthFilter(maxSubjectLength ?: 150))
```
```kotlin
  private fun getMaxSubjectLength(): Int? = Snabble.checkedInProject.value
        ?.id
        ?.let { id ->
            Snabble.customProperties
                .getOrDefault(
                    ExternalBillingSubjectLength to ProjectId(id),
                    defaultValue = null
                )
                ?.toString()
                ?.toInt()
        }
```

**Review by Fabtron — CHANGES_REQUESTED** [review]
Works good just some small change as suggestion

File: ui/src/main/java/io/snabble/sdk/ui/payment/externalbilling/ui/widgets/SubjectAlertDialog.kt (hunk 1 above)
Comment (cmaier) [reply ui/src/main/java/io/snabble/sdk/ui/payment/externalbilling/ui/widgets/SubjectAlertDialog.kt]: Done!

**Review by Fabtron — APPROVED** [review]
Nice
